src/sagemaker_server.cc

```diff
@@ -670,6 +732,35 @@ SagemakerAPIServer::SageMakerMMEUnloadModel(
 
   HandleRepositoryControl(req, "", model_name, "unload");
 
+  /*Note: Model status check is repo-specific and therefore must be run before
+   * unregistering the repo, else the model information is lost*/
+  bool is_model_unavailable = false;
+  uint32_t unload_time_in_secs = 0;
+
+  LOG_VERBOSE(1) << "Using Model Repository Index during UNLOAD to check for "
+                    "status of model: "
+                 << model_name;
+
+  /* Wait for the model to be completely unloaded. SageMaker waits a maximum of
+  360 seconds for the UNLOAD request to timeout. Setting a limit of 350 seconds
+  for Triton unload.*/
+  while (!is_model_unavailable && unload_time_in_secs < UNLOAD_TIMEOUT_SECS_) {
+    is_model_unavailable = SageMakerMMEUnloadModelCheckStatus(model_name);
+    sleep(1);
```

Comment:
The sleep amount could also be a constant, like `UNLOAD_TIMEOUT_SECS_` is. Could also use `std::this_thread::sleep_for(std::chrono::milliseconds(UNLOAD_SLEEP_MILLISECONDS_));` instead of `sleep` to get more granularity, if desired (especially if you frequently need more exactly one sleep but it's usually done much more quickly than one second).

Reply:
Thanks, used the `std::this_thread::sleep_for(
          std::chrono::milliseconds(UNLOAD_SLEEP_MILLISECONDS_))` to sleep for a shorter duration